### PR TITLE
Add cmake-mingw in circle-build

### PR DESCRIPTION
**Pull request by akankshamahajan15**
Summary: Add cmake-mignw in circle-build

Test Plan: watch circle cmake-mingw build

Reviewers:

Subscribers:

Tasks: T70028805

Tags:

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -364,6 +364,29 @@ jobs:
...
+          command: |
+            sudo apt-get install snapd && sudo snap install cmake --beta --classic
+            export PATH=/snap/bin:$PATH
+            sudo apt-get install -y openjdk-8-jdk
+            export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
+            export PATH=$JAVA_HOME/bin:$PATH
+            echo "JAVA_HOME=${JAVA_HOME}"
+            which java && java -version
+            which javac && javac -version
+            mkdir build && cd build && cmake -DJNI=1 -DWITH_GFLAGS=OFF .. -DCMAKE_C_COMPILER=x86_64-w64-mingw32-gcc -DCMAKE_CXX_COMPILER=x86_64-w64-mingw32-g++ -DCMAKE_SYSTEM_NAME=Windows && make -j4 rocksdb rocksdbjni
```
Comment (jay-zhuang) [inline .circleci/config.yml:387]: do we need to run test?

File: .circleci/config.yml [hunk 2]
```diff
@@ -354,7 +354,7 @@ jobs:
   build-examples:
     machine:
       image: ubuntu-1604:202007-01
-    resource_class: medium
+    resource_class: large
```
Comment (jay-zhuang) [inline .circleci/config.yml:357]: is medium too slow?

**Review by jay-zhuang — APPROVED** [review]
LGTM. Thanks.

File: .circleci/config.yml (hunk 2 above)
Comment (akankshamahajan15) [reply .circleci/config.yml:357]: yes it is. Thats why i changed it to large

File: .circleci/config.yml (hunk 1 above)
Comment (akankshamahajan15) [reply .circleci/config.yml:387]: In travis, these tests were run so added the same command here as well.

**Pull request closed without merging**